controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -64,6 +65,40 @@ func (in *KubeadmControlPlane) Default() {
 	if !strings.HasPrefix(in.Spec.Version, "v") {
 		in.Spec.Version = "v" + in.Spec.Version
 	}
+
+	ios1 := intstr.FromInt(1)
+	ios0 := intstr.FromInt(0)
+	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
+	if in.Spec.RolloutStrategy != nil {
+		if in.Spec.RolloutStrategy.Type != RollingUpdateStrategyType {
+			in.Spec.RolloutStrategy.Type = RollingUpdateStrategyType
+		}
+
+		// Use default values if user has not set RoloutStrategy.RollingUpdate
+		if in.Spec.RolloutStrategy.RollingUpdate == nil {
+			in.Spec.RolloutStrategy.RollingUpdate = &RollingUpdate{}
+			in.Spec.RolloutStrategy.RollingUpdate.MaxSurge = &ios1
+			in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable = &ios0
+		} else {
+			// Make sure to default with right values if the user has
+			// set only Spec.RolloutStrategy.RollingUpdate.MaxSurge or
+			// Spec.RolloutStrategy.RollingUpdate.MaxUnavailable
+			if in.Spec.RolloutStrategy.RollingUpdate.MaxSurge == nil && in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable != nil {
+				if *in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable == ios0 {
+					in.Spec.RolloutStrategy.RollingUpdate.MaxSurge = &ios1
+				} else {
+					in.Spec.RolloutStrategy.RollingUpdate.MaxSurge = &ios0
+				}
+			}
+			if in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable == nil && in.Spec.RolloutStrategy.RollingUpdate.MaxSurge != nil {
+				if *in.Spec.RolloutStrategy.RollingUpdate.MaxSurge == ios1 {
+					in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable = &ios0
+				} else {
+					in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable = &ios1
+				}
+			}
+		}
+	}
```

Comment:
I think this can be simplified down a bit to avoid as much branching.

This changes the default behavior, but it stays in line with what we are calling out in the type documentation, though.

```suggestion
		if in.Spec.RolloutStrategy.RollingUpdate == nil {
			in.Spec.RolloutStrategy.RollingUpdate = &RollingUpdate{}
		}

		in.Spec.RolloutStrategy.RollingUpdate.MaxSurge == intstr.ValueOrDefault(in.Spec.RolloutStrategy.RollingUpdate.MaxSurge, ios1)
		
		in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable == intstr.ValueOrDefault(in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable, ios0)
		
```